### PR TITLE
Methodology page

### PR DESCRIPTION
**Pull request by Bluesmile82**
https://vizzuality.atlassian.net/browse/ORC-331?atlOrigin=eyJpIjoiODM5NmQ2MTQwY2NkNDE5NmFkNDc2YWE0MjcwNGNhYmEiLCJwIjoiaiJ9

Added methodology page and chart. Open it by clicking on the methodology button.

Missing one of the links to the emails

### CODE REVIEW

**Comment by clementprdhomme** [comment]
The missing email address is: damien.beillouin@cirad.fr

File: javascript/api-service.js [hunk 1]
```diff
@@ -153,6 +153,36 @@ const getPublications = async ({ landUse, mainIntervention, intervention, subTyp
...
+    publications: publicationYearCounts,
+  };
+};
+
+const getMethodologyData = async () => {
+  const url = getPathname(BASE_PUBLICATIONS_URL, 'all');
+  return getURL(url).then(data => {
+    return getMethodologyYears(data);
+  });
+}
```
Comment (clementprdhomme) [inline javascript/api-service.js:184]: François has probably paginated this endpoint. Could you use `getURLParams` and  pass a `pageSize` parameter as we do in `getPublications`?

Have you talked to him before making this implementation? We're loading *all* the publications to render this graph which may not be performant at all.

File: javascript/api-service.js [hunk 2]
```diff
@@ -153,6 +153,36 @@ const getPublications = async ({ landUse, mainIntervention, intervention, subTyp
...
+  const publicationYearCounts = {};
+  data.concat().forEach(publication => {
+    if (publication.type === 'meta-analysis') {
+      const year = publication.year;
+      metaAnalysisYearCounts[year] = (metaAnalysisYearCounts[year] || 0) + 1;
+    }
+    if (publication.type === 'primary-paper') {
+      const year = publication.year;
+      publicationYearCounts[year] = (publicationYearCounts[year] || 0) + 1;
+    }
```
Comment (clementprdhomme) [inline javascript/api-service.js:170]: Why are you adding 1 to `publicationYearCounts`?

File: javascript/d3-methodology-chart.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    .domain([0, Math.max(...publicationsData.map(d => d.value))])
+    .range([ height - VERTICAL_AXIS_PADDING, 0 ]);
+
+  const TICK_DISTANCE = 500;
+  const numTicks = Math.ceil((y.domain()[1] - y.domain()[0]) / TICK_DISTANCE);
+  const tickValues = Array.from({length: numTicks}, (_, i) => {
+    return y.domain()[0] + i * TICK_DISTANCE;
+  });
+
+  const yAxis = d3.axisLeft(y)
```
Comment (clementprdhomme) [inline javascript/d3-methodology-chart.js:64]: The Y axis is not rendering.

File: index.html [hunk 4]
```diff
@@ -211,6 +219,112 @@ <h2 class="leading-[18px] text-lg"><span
...
+              reach a global overview of the impacts of multiple interventions
+              on Soil Organic Carbon, with high statistical power. We also
+              characterized the methodological quality of the meta-analyses, and
+              listed all the primary studies synthethized in the meta-analyses
+              to clearly identify and present potential bias and limitation in
+              the current scientific knowledge.</div>
+            <div class="flex flex-wrap">
+              <div class="flex-grow"
+                id="methodology-chart"></div>
+              <div id="methodology-legend">
```
Comment (clementprdhomme) [inline index.html:264]: On Chrome on my computer, the legend is displayed below the graph. It does not happen on Firefox.

**Review by clementprdhomme — COMMENTED** [review]
In addition to my inline comments, I have one question for you.

You said to Carla that we cannot change the map's legend because the markers represent a sum of publications and not a number of primary studies. Nevertheless, [this code here](https://github.com/Orcasa-Platform/orcasa-review4c/blob/main/javascript/map/map-layers.js#L111-L116) seems to indicate that we're displaying primary studies. If that's the case, could you update the legend then?

File: javascript/api-service.js (hunk 1 above)
Comment (Bluesmile82) [reply javascript/api-service.js:184]: I'm going to send a message in basecamp. This should ideally be a different endpoint and this logic should be on the backend. Maybe I'll just comment this code and add a mock for it

File: javascript/api-service.js (hunk 2 above)
Comment (Bluesmile82) [reply javascript/api-service.js:170]: This is to count the meta-analyses or primary paper. On each instance of the data we add 1 to the count

**Comment by Bluesmile82** [comment]
I added a change on the chart to show in the line when we have years with no publications. I'm not sure if this will happen on the final data but just to be sure to display correct data

File: index.html (hunk 4 above)
Comment (clementprdhomme) [reply index.html:264]: No it's fine. It's just that with the same screen size, on Firefox it's on the side and on Chrome below. Not a problem.

File: javascript/api-service.js (hunk 2 above)
Comment (clementprdhomme) [reply javascript/api-service.js]: Oh ok 🤦.

**Review by clementprdhomme — COMMENTED** [review]
I think that the only thing remaining is the change to the map legend that Carla asked about.

**Comment by Bluesmile82** [comment]
Ah, sorry, there is another task for that as it required a better specification: https://vizzuality.atlassian.net/browse/ORC-382?atlOrigin=eyJpIjoiY2I5OTNiOGJmNWFkNDAxZjk0YTc1N2E1Y2JhNmZmMjMiLCJwIjoiaiJ9

**Pull request merged**